### PR TITLE
add snow and ice stairs and slabs

### PR DESCRIPTION
**Pull request by timsoftgit**
this adds snow and ice stairs and slabs, useful for ice castles/palaces, paths etc.

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -525,7 +525,7 @@ stairs.register_stair_and_slab(
 )
 
 stairs.register_stair_and_slab(
-	"obsidian_block",
+ 	"obsidian_block",
```
Comment (sfan5) [inline mods/stairs/init.lua]: accidental space

File: mods/stairs/init.lua [hunk 2]
```diff
@@ -593,3 +593,22 @@ stairs.register_stair_and_slab(
 	"Gold Block Slab",
 	default.node_sound_metal_defaults()
 )
+stairs.register_stair_and_slab(
```
Comment (sfan5) [inline mods/stairs/init.lua]: empty line missing here

File: mods/stairs/init.lua [hunk 3]
```diff
@@ -593,3 +593,22 @@ stairs.register_stair_and_slab(
...
+	"ice",
+	"default:ice",
+	{cracky = 3, puts_out_fire = 1},
```
Comment (sfan5) [inline mods/stairs/init.lua]: Maybe this should have `cools_lava = 1` too?

[see here](https://github.com/minetest/minetest_game/blob/master/mods/default/nodes.lua#L557)

File: mods/stairs/init.lua [hunk 4]
```diff
@@ -593,3 +593,22 @@ stairs.register_stair_and_slab(
...
+	{"default_ice.png"},
+	"Ice Stair",
+	"Ice Slab",
+	default.node_sound_glass_defaults()
+)
+
+stairs.register_stair_and_slab(
+	"snow",
+	"default:snow",
+	{crumbly = 3, puts_out_fire = 1},
```
Comment (sfan5) [inline mods/stairs/init.lua]: `snowy = 1` missing

Comment (timsoftgit) [inline mods/stairs/init.lua]: also added in update

File: mods/stairs/init.lua (hunk 3 above)
Comment (timsoftgit) [inline mods/stairs/init.lua]: added in update

File: mods/stairs/init.lua (hunk 2 above)
Comment (timsoftgit) [inline mods/stairs/init.lua]: fixed in update

File: mods/stairs/init.lua (hunk 1 above)
Comment (timsoftgit) [inline mods/stairs/init.lua]: fixed in update

**Comment by timsoftgit** [comment]
hi sfan5, is the pull request ok now, or do i need to do anything else?

**Comment by paramat** [comment]
Hmm ice seems ok but snow .. if we allow snow then we would have to do other soft nodes like sands, dirt etc. So i'm unsure about snow.

**Comment by ExeterDad** [comment]
I'm okay with snow stairs. Snow can definitely be compact enough to support weight. I've decades of winter experience (living in the North) to vouch for that. Igloo's for example are very durable and are made from compacted snow.
Why else would a snow block include the word  "block?"  I'm of the opinion that snow stairs would be welcomed, and even have been asked if they exist or are possible to add to our server.

**Comment by sofar** [comment]
This should use snow sounds like the snow block, not pure dirt sounds:

https://github.com/minetest/minetest_game/blob/master/mods/default/nodes.lua#L520

**Comment by paramat** [comment]
^ Yes, i missed that.

**Comment by paramat** [comment]
@timsoftgit sounds need correcting for snow.

**Comment by paramat** [comment]
@timsoftgit

**Comment by tenplus1** [comment]
I would have preferred snow and ice bricks to be added and stairs for those, like I do in ethereal mod.

**Comment by Ferk** [comment]
With snowbricks it would be possible to make (cubic) igloos, and it would be consistent with soft sand nodes making sandstone bricks.

**Comment by paramat** [comment]
That could be another PR, the basic nodes are needed first though.

**Comment by paramat** [comment]
Sounds can be corrected on merge. Author is absent.

**Comment by paramat** [comment]
:+1: as i will correct on merge. SmallJoker and sofar also support.

**Comment by paramat** [comment]
519d166ce34d5516b0b20dd86ca0bf5ce5bf658c
Also changed 'snow' to 'snowblock' and changed groups accordingly, did not see this error earlier.

**Comment by timsoftgit** [comment]
I don't seem to be getting notifications from git, so sorry if I have been "quiet"  but is any more input required, or is this going through (in one form or another)?

**Comment by paramat** [comment]
Ok no problem. It's already merged.

**Pull request closed without merging**